scripts/test-dev.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+TEST_SUBDIR_PENDING="true"
+TEST_NAME_PENDING="true"
+
+# Flag: dev mode for testing (and not usual CI)
+export STARKNET_HARDHAT_DEV=1
+export STARKNET_HARDHAT_DEV_NETWORK="integrated-devnet"
+
+cd test
+
+# Loops until a suitable TEST_SUBDIR is provided
+while [[ -n $TEST_SUBDIR_PENDING ]]; do
+    echo "(Tab to autocomplete)"
+    read -e -p "Test suite:" TEST_SUBDIR
+    TEST_SUBDIR="${TEST_SUBDIR%/}"
+
+    if [[ -d $TEST_SUBDIR ]]; then
+        if [[ $TEST_SUBDIR != "${TEST_SUBDIR%-tests}" ]]; then
+            TEST_SUBDIR_PENDING=""
+        fi
+    else
+        echo ""
+        echo "Please pick from,"
+        ls | grep '\-tests' | awk '{print " - "$1}'
+        echo ""
+    fi
+done
+
+TEST_SUBDIR="${TEST_SUBDIR%/}" # remove trailing slash
+
+cd $TEST_SUBDIR
+
+# Loops until a suitable test_name is provided
+while [[ -n $TEST_NAME_PENDING ]]; do
+    echo ""
+    echo "(Tab to autocomplete)"
+    read -e -p "Test:" test_name
+    test_name="${test_name%/}"
+
+    if [[ -d $test_name ]]; then
+        echo "Running test $test_name from $TEST_SUBDIR"
+        TEST_NAME_PENDING=""
+    else
+        echo "Please pick from,"
+        ls | awk '{print " - "$1}'
+        echo ""
+    fi
+done
+
+cd ../..
+
+test_name="${test_name%/}" # remove trailing slash
+RUN_SETUP="y"
+CONFIG_FILE_NAME="hardhat.config.ts"
+
+if [[ -d starknet-hardhat-example/ ]]; then
+    echo ""
+    read -e -p "Example repo found, y to force run setup: " RUN_SETUP
+fi
+
+if [[ -n $RUN_SETUP ]]; then
+    echo ""
+    source  ./scripts/setup-cairo1-compiler.sh
+    rm -rf starknet-hardhat-example
+    git clone -b "${EXAMPLE_REPO_BRANCH:=plugin}" --single-branch https://github.com/0xSpaceShard/starknet-hardhat-example.git
+    cd starknet-hardhat-example
+    npm ci
+    npm install ../ # install plugin from source (parent dir)
+    cd ..
+else
+    echo "Skipped setup."
+fi
+
+pwd
+
+echo ""
+echo "If your test needs devnet (and not integrated-devnet),"
```

Comment:
Can't we guess the need for devnet/integrated-devnet by looking into network.json?

Reply:
It looked like by design tests could have multiple networks. So I thought it would be easy to just give the devs a choice. Since it's just a one time thing seemed to be okay.
Let me know what you think ;)

Reply:
ok then